sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_file_client.py

```diff
@@ -597,6 +597,49 @@ def download_file(self, offset=None, length=None, **kwargs):
         downloader = self._blob_client.download_blob(offset=offset, length=length, **kwargs)
         return StorageStreamDownloader(downloader)
 
+    def create_snapshot(self, metadata=None, **kwargs):
```

Comment:
we need to support download snapshot get snapshot properties also, else user cannot use the created snapshot using our sdk

Reply:
Done